charts/patch/templates/patch.json

```diff
@@ -44,7 +45,7 @@
   },
   {{- end }}
   {{- end}}
-  {{- if .Values.global.proxyInit }}
+  {{- if and .Values.global.proxyInit (not .Values.global.cniEnabled) }}
```

Comment:
Same here, as `.Values.global.proxyInit` has values, even when the installation is through CNI, and hence we need to depend on `.global.cniEnabled` to skip InitContainer INjection